shapes/contacts/person.shex

```diff
@@ -0,0 +1,47 @@
+PREFIX vcard: <http://www.w3.org/2006/vcard/ns#>
+PREFIX xsd:   <http://www.w3.org/2001/XMLSchema#>
+PREFIX owl: <http://www.w3.org/2002/07/owl#>
+PREFIX dc: <https://dublincore.org/specifications/dublin-core/dcmi-terms/#>
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+
+# We should store the preferred WebId for each person, since sameAs can be multiple WebIds
+
+vcard:Individual {
+    vcard:fn xsd:string {1} ; 
+    vcard:inAddressBook vcard:AddressBook + ;
+    owl:sameAs vcard:Individual * ; # WebIds
+    vcard:hasUID xsd:string ? ;
+    vcard:hasName xsd:string ? ; # structured name 
+    vcard:hasPhoto dc:Image * ; 
+    vcard:hasRelated vcard:RelatedType * ; 
+    vcard:url rdf:Resource * ; # need to describe the type of url to be able to identify WebIds 
+    vcard:hasAddress vcard:Address * ; 
+    vcard:bday xsd:date ? ; 
+    vcard:hasEmail vcard:Email * ; 
+    vcard:hasTelephone vcard:Tel * ; 
+    vcard:organization-name xsd:string ? ; 
+    vcard:role xsd:string ? ; 
+    vcard:title xsd:string ? ; 
+    vcard:note xsd:string ? ;
+}
+
+vcard:Address {
```

Comment:
Would like to include Address type similar to email and phone types.

While not explicitly needed, may be worth putting min and max string lengths in the shape to stop users entering novels in the fields.

Reply:
Another question: what do you think would be a good upper limit? The string length limit would be for:

- Street name
- Locality
- Region
- Country name

Some street names and city names can be very long in some parts of the world: https://en.wikipedia.org/wiki/List_of_long_place_names

Reply:
@timbl did not want min and max string lengths in the shape - he argued it should be specific to the the form models, not the shape itself.

Reply:
What do you mean with `Would like to include Address type similar to email and phone types.` btw? I think @VirginiaBalseiro has addressed that?

Reply:
I did address the types, this comment was still unresolved due to the max length question but the comment is prior to my latest changes where I added types to Address, email and Tel.

Reply:
Since Tim said he didn't want the MAXLENGTH and MINLENGTH, I'll resolve this thread.

Reply:
We may want to give this some more thought. Originally, the React SDK v1 did not use the min and max lengths. However, when the team implemented the ShEx Shape -> Form Model  Converter, the min and max lengths (along with other field parameters) were _very_ useful to help select the Form Model field type to use for each of the ShEx shape fields. For example, based on the string max length, the Converter could decide whether to use Textboxes or Text Areas. Similar with Integer fields to decide whether it should use a pre-populated dropdown (a few integers), a slider, or Textbox.

Reply:
Re: Address Type. If @VirginiaBalseiro has added one, then great.

Reply:
One thing to consider with the Address is the fact there are only one Street Name and one Locality. In way of an example, in England, locals like naming their houses and having multiple locales. When I was growing up, my official address was:

- The Bungalow, (Name of house)
- River Road, 
- West Walton (Village)
- Wisbech (Town)
- Cambs (County)
- PE14 7EX (Post Code)
- UK

I have similar issues with family and friends, and I am sure this is not unique to the UK.

If we want to be sophisticated, we could enhance the shape to allow for these kinds of addresses.

Reply:
We could change it to `vcard:street-address xsd:string + ;` which would allow for one or more street addresses? (And same for `vcard:locality xsd:string +;`.)

Reply:
We should use whatever DB uses for now and continue the conversation about what's optimal in a separate ticket.